subiquity/tests/api/test_api.py

```diff
@@ -247,7 +247,7 @@ async def test_server_flow(self):
             resp = await inst.get('/storage/guided')
             disk_id = resp['disks'][0]['id']
             choice = {"disk_id": disk_id}
-            await inst.post('/storage/v2/guided', choice)
+            await inst.post('/storage/v2/deprecated/guided', choice)
```

Comment:
I presume you'll write some api tests for the new API at some point?

Reply:
Thanks, I meant to do that.
See new commits.
I lean a little bit too much on the API tests so I made a point of doing similarish things in the unit test layer, and I'm happy with the result, but a few at the API layer are a good idea.